src/GT.Core/Services/Impl/GTListingService.cs

```diff
@@ -1,9 +1,277 @@
+			{
+				_logger.LogError($"Attempted to add a null reference to the database.");
+				return null;
+			}
+
+			// Map DTO to entity
+
+			var newListing = await CreateListingEntity(listingDTO, signedInUserId);
+
+			// Update database
+			await _listingRepository.AddAsync(newListing);
+
+
+			// Map updates to DTO
+			listingDTO.Id = newListing.Id;
+
+			// return the updated ListingDTO
+			return listingDTO;
+		}
+
+		public async Task<Listing> CreateListingEntity(ListingDTO listingDTO, string signedInUserId)
+		{
+			var newListing = new Listing()
+			{
+			};
+			newListing.Id = Guid.NewGuid().ToString();
+			newListing.ListingTitle = listingDTO.ListingTitle;
+			newListing.Description = listingDTO.Description;
+			newListing.SalaryMin = listingDTO.SalaryMin;
+			newListing.SalaryMax = listingDTO.SalaryMax;
+			newListing.JobTitle = listingDTO.JobTitle;
+			newListing.FTE = listingDTO.FTE;
+			newListing.CreatedDate = listingDTO.CreatedDate;
+			newListing.CreatedById = signedInUserId;
+
+
+			// Check if company exists
+			if (listingDTO.Employer is not null)
+			{
+				// TODO if exists 
+				var employer = await _companyRepository.GetAll().Where(e => e.Name == listingDTO.Employer).FirstOrDefaultAsync();
+				if (employer is null)
+				{
+					await _companyService.AddAsync(new CompanyDTO() { Name = listingDTO.Employer });
+				}
+
+				newListing.Employer = await _companyRepository.GetAll().Where(e => e.Name == listingDTO.Employer).FirstOrDefaultAsync();
+			}
+
+
+			// Check if location exists
+			if (listingDTO.Location is not null)
+			{
+				var location = await _locationRepository.GetAll().Where(e => e.Name == listingDTO.Location).FirstOrDefaultAsync();
+				if (location is null)
+				{
+					await _locationService.AddAsync(new LocationDTO() { Name = listingDTO.Location });
+				}
+				newListing.Location = await _locationRepository.GetAll().Where(e => e.Name == listingDTO.Location).FirstOrDefaultAsync();
+			}
+
+
+			// Check if experienceLevel exists
+			var experienceLevel = await _experienceLevelRepository.GetAll().Where(e => e.Name == listingDTO.ExperienceLevel).FirstOrDefaultAsync();
+			if (experienceLevel is null)
+			{
+				// TODO - ExperienceLevel does not exist and need to be created.
+			}
+			newListing.ExperienceLevel = experienceLevel;
+
+
+			return newListing;
+		}
+
+		// TODO - Should this return anything if string id is not found? Return a delete success/fail bool?
+		public async Task DeleteAsync(string id)
+		{
+			if (_listingRepository.GetAll().Any(e => e.Id == id))
+			{
+				await _listingRepository.DeleteAsync(id);
+			}
+		}
+
+		public async Task<List<ListingPartialDTO>> GetAsync(IListingFilterModel? filter = null)
+		{
+			// Gets entities from database
+			if (filter is null)
+			{
+				filter = new ListingFilterModel();
+			}
+
+			var query = _listingRepository
+				.GetAll()
+				.Include(e => e.ExperienceLevel)
+				.Include(e => e.Location)
+
+				.Where(e => filter.ExperienceLevels == null
```

Comment:
Lambda vs Ternary - (applies to all filters below) ?

Reply:
This will be managed in issue #64.